vm: make the `nimVMDebugGenerate` define work again

zerbina opened the pull request:
## Summary
Fix the compilation error in `vmjit` that happened when
`debugEchoCode` was enabled.

In addition, refactor, clean-up, and document the `codeListing`
procedure and move it to the new module `vmutils`. `codeListing` also
doesn't directly report the generated listing anymore - that's the
responsibility of it's caller now.

## Details
* the `debugEchoCode` constant is removed and replaced with
 direct usage of `defined(nimVMDebugGenerate)`

haxscramper commented on compiler/vm/vm.nim:
```diff
@@ -87,7 +87,8 @@ const
 
-
+when debugEchoCode:
+  import compiler/vm/vmutils
```
Ideally, conditional imports should be placed consistently with the rest of imports - that is, at the start of the file, not before the first usage.

haxscramper commented on compiler/vm/vmjit.nim:
```diff
@@ -38,6 +38,10 @@ export VmGenResult
 const
   debugEchoCode* = defined(nimVMDebugGenerate)
 
+when debugEchoCode:
```
Preferably should be moved to the top of the file as well.

zerbina replied:
That's only possible if the usage of `debugEchoCode` here is replaced with `defined(nimVMDebugGenerate)`, which would mean that the imports are guarded by a different expression than the dependents of the import, rendering `debugEchoCode` somewhat useless.

I can move the `debugEchoCode` constant to `vmdef` instead.

haxscramper replied:
Just remove `debugEchoCode` altogether and use defines as the rest of debugging utils does

haxscramper reviewed (approved):
`debugEchoCode` define is currently missing from the documentation, so if you can, please do add it (at least in the "Debugging Defines - Quick Reference" section) in `debug.rst`. Aside from minor comments on the import placement I don't have anything to say, and since this PR does not seem to change the behavior in any way I think it can be merged.

zerbina commented:
> `debugEchoCode` define is currently missing from the documentation, so if you can, please do add it (at least in the "Debugging Defines - Quick Reference" section) in `debug.rst`.

Hm, but what for? It's not something that can be separately toggled, as it's just an alias for `defined(nimVMDebugGenerate)`.

zerbina committed:
vm: make the `nimVMDebugGenerate` define work again

## Summary
Fix the compilation error in `vmjit` that happened when
`debugEchoCode` was enabled.

In addition, refactor, clean-up, and document the `codeListing`
procedure and move it to the new module `vmutils`. `codeListing` also
doesn't directly report the generated listing anymore - that's the
responsibility of it's caller now.

## Details
* the `debugEchoCode` constant is removed and replaced with
 direct usage of `defined(nimVMDebugGenerate)`

zerbina commented:
As requested, I removed `debugEchoCode` and replaced it's usage with `defined(nimVMDebugGenerate)`.

bors r+
